### PR TITLE
front: refacto infras list module

### LINKED ISSUE #4679: front: refacto and migrate infras list editor module
### Description and goal

The current clone infra function is working but has been patched with a wrong behaviour, it need to be refactored as well as the rest of the module which need to be migrate in tsx and rtk.

### Acceptance criteria

- [ ] after an infra clone, the infras list should be refreshed without setTimeout
- [ ] renaming an infra should use the rtk editoast endpoint and the list should be properly refreshed
- [ ] locking an infra should use the rtk editoast endpoint and the list should be properly refreshed
- [ ] deleting an infra should use the rtk editoast endpoint and the list should be properly refreshed
- [ ] exporting an infra should use the rtk editoast endpoint
- [ ] all jsx files of the module should be migrated in tsx
- [ ] adding an infra should be still working properly

### LINKED ISSUE #4683: List of infra limited to 25 entries
### What happened?

When I add infrastructures, after there are 25 of them, they do not appear in the infrastructure selection tool

### What did you expect to happen?

The full list of infrastructures should be visible

### How can we reproduce it (as minimally and precisely as possible)?

1. Go to cartography
2. Click on infra selection tool
3. Click on edition button
4. Delete infrastructures (the "test x" ones for example) until there are less than 25
5. Add some infras and witness that the list says "25 infrastructure found" regardless of the number of infras you may add

### What operating system, browser and environment are you using?

- **Browser**: Firefox v109
- **OS**: Ubuntu
- **Env**: dev

### OSRD version (top right corner `Account` button > `Informations`)

220bd545

### PR DESCRIPTION
**Pull request by SharglutDev**
close #4679 
close #4683

### CODE REVIEW

**Review by Tguisnet — APPROVED** [review]

File: front/src/common/InfraSelector/InfraSelectorEditionActionsBar.tsx [hunk 1]
```diff
@@ -1,40 +1,51 @@
...
-        getInfrasList();
-        setIsWaiting(false);
+        if (action === InfraLockState.LOCK) {
+          await lockInfra({ id: infra.id });
+        }
+        if (action === InfraLockState.UNLOCK) {
+          await unlockInfra({ id: infra.id });
+        }
       } catch (e) {
+        if (e instanceof Error) {
```
Comment (Uriel-Sautron) [inline front/src/common/InfraSelector/InfraSelectorEditionActionsBar.tsx:40]: We're sure that e will always be an instance of Error?

File: front/src/common/InfraSelector/InfraSelectorEditionActionsBar.tsx [hunk 2]
```diff
@@ -44,10 +55,18 @@ export default function ActionsBar({
     if (!isWaiting) {
       setIsWaiting(true);
       try {
-        const railjson = await get(`${INFRA_URL}${infra.id}/railjson/`);
+        const railjson = await getRailjson({ id: infra.id });
         fileDownload(JSON.stringify(railjson), `${infra.name}.id${infra.id}.railjson.json`);
-        setIsWaiting(false);
       } catch (e) {
+        if (e instanceof Error) {
```
Comment (Uriel-Sautron) [inline front/src/common/InfraSelector/InfraSelectorEditionActionsBar.tsx:61]: Same thing here

**Review by Uriel-Sautron — COMMENTED** [review]
Very good work, tested on local

File: front/src/common/InfraSelector/InfraSelectorEditionActionsBar.tsx (hunk 1 above)
Comment (SharglutDev) [reply front/src/common/InfraSelector/InfraSelectorEditionActionsBar.tsx:40]: Couldn't test it but it's coded like that in other components so I thought it was ok and still better to have a toast displayed to the user if an action fails.

File: front/src/common/InfraSelector/InfraSelectorEditionActionsBar.tsx (hunk 2 above)
Comment (SharglutDev) [reply front/src/common/InfraSelector/InfraSelectorEditionActionsBar.tsx:61]: Same as above

File: front/src/common/InfraSelector/InfraSelectorEditionActionsBar.tsx (hunk 1 above)
Comment (SharglutDev) [reply front/src/common/InfraSelector/InfraSelectorEditionActionsBar.tsx:40]: Ok I found something, let me know if you are ok

**Review by Math-R — APPROVED** [review]

**Pull request merged**